subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformWithDependenciesIntegrationTest.groovy

```diff
@@ -325,4 +329,220 @@ project(':app') {
         outputLines.any { it ==~ /TestTransform: .*slf4j-api-1.7.25.jar is not up-to-date because:/ }
         outputLines.any { it == "Single step transform received dependencies files [slf4j-api-1.7.25.jar, common.jar] for processing lib.jar" }
     }
+
+    def "transforms with different dependencies in multiple dependency graphs are executed"() {
+        given:
+        mavenHttpRepo.module("org.slf4j", "slf4j-api", "1.7.26").publish().allowAll()
+        settingsFile << "include('app2')"
+        buildFile << """
+            project(':app') {
+                task resolveSize(type: Copy) {
+                    def artifacts = configurations.implementation.incoming.artifactView {
+                        attributes { it.attribute(artifactType, 'size') }
+                    }.artifacts
+                    from artifacts.artifactFiles
+                    into "\${buildDir}/libs/size"
+                }
+                task resolveInter(type: Copy) {
+                    def artifacts = configurations.implementation.incoming.artifactView {
+                        attributes { it.attribute(artifactType, 'inter') }
+                    }.artifacts
+                    from artifacts.artifactFiles
+                    into "\${buildDir}/libs/inter"
+                }         
+            }
+            project(':app2') {
+                task resolveSize(type: Copy) {
+                    def artifacts = configurations.implementation.incoming.artifactView {
+                        attributes { it.attribute(artifactType, 'size') }
+                    }.artifacts
+                    from artifacts.artifactFiles
+                    into "\${buildDir}/libs/size"
+                }
+                task resolveFinal(type: Copy) {
+                    def artifacts = configurations.implementation.incoming.artifactView {
+                        attributes { it.attribute(artifactType, 'final') }
+                    }.artifacts
+                    from artifacts.artifactFiles
+                    into "\${buildDir}/libs/final"
+                }         
+                dependencies {
+                    implementation 'junit:junit:4.11'
+                    implementation 'org.slf4j:slf4j-api:1.7.26'
+                    implementation project(':lib')
+                }
+            }
+        """
+
+        when:
+        run "resolveSize", "resolveInter", "resolveFinal"
+        then:
+        output.count("Transforming") == 20
```

Comment:
It would be nice to use something more unique here. For example, we also log this string at `INFO` level, but I take this is the string printed by `SimpleTransform` that we are looking for here.

Reply:
Also, having the number 20 without context seems brittle and doesn't explain what's happening too well. It would be great if we could expect something more specific. I think even 20 lines of actual expectations written out would be better.